gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableFile.java

```diff
@@ -68,6 +78,83 @@
   /** Checksum of the origin file. */
   private byte[] checksum;
 
+  public static Builder builder(FileSystem originFs, FileStatus origin, Path targetRoot, Path datasetRoot) {
```

Comment:
I think this defeats the purpose of having a builder class, which is to avoid having constructor or factory class that have too many arguments. Why not simply have a method that returns a plain `Builder` instance, like `public static Builder builder`?